src/Streams.Core/Streams.fs

```diff
@@ -4,20 +4,18 @@ open System.Collections.Generic
 open System.Threading
 
 /// Provides on-demand iteration 
-type Iterator = {
+type (* internal *) Iterator = 
     /// Function for on-demand processing
-    TryAdvance : unit -> bool 
+    abstract TryAdvance : unit -> bool 
     /// Cleanup function
-    Dispose : unit -> unit 
-}
+    abstract Dispose : unit -> unit 
 
 /// Provides functions for iteration
-type Iterable = {
+type (* internal *) Iterable = 
```

Comment:
(It looks like Dispose always needs to be called - or at least I suspect the current implementation is leaky in the sense that there are cases where the Iterable gets consumed via Bulk() but an iterator has been allocated for the object - though not used)